src/index.ts

```diff
@@ -1250,16 +1289,15 @@ class Envoy {
     let retryCall = true
     while (retryCall) {
       retryCall = false
-
       try {
-        resp = await this.callConductor("app", {
+        resp = await this.callSlUpdate({
```

Comment:
This looks pretty good, but I can't quite tell what the consequences are of a source-chain-head moved error to this code.  It looks like `callSlUpdate` just queues the request.  Will the queuing of the request ever return the moved error here?  I'm also noting that other calls to `callSlUpdate` (i.e. the disk space one) aren't checking for the source chain head moved for retry.  Should that test be moved elsewhere?

Reply:
We don't expect source chain head moved to ever occur. That's the point of the queuing mechanism. `callSlUpdate` queues the request and then waits on a callback that the request has either resolved or rejected so if the conductor call throws source chain head moved then so will `callSlUpdate`. We didn't bother updating the disk space case because it's not important if we miss one of those logs and because it shouldn't ever happen anyway,